crates/dojo-core/src/world.cairo

```diff
@@ -250,16 +267,11 @@ mod world {
         fn register_component(ref self: ContractState, class_hash: ClassHash) {
             let caller = get_caller_address();
             let calldata = ArrayTrait::new();
-            let name = *self
-                .executor_dispatcher
-                .read()
-                .call(class_hash, NAME_ENTRYPOINT, calldata.span())[0];
+            let name = *_class_call(class_hash, NAME_ENTRYPOINT, calldata.span())[0];
```

Comment:
we can't do this. we need to execute through the executor so the call is sandboxed. since it is executing untrusted code, the component implementation could modify world storage directly

Reply:
Aah, true that...